src/clr-angular/forms/textarea/textarea-container.spec.ts

```diff
@@ -0,0 +1,161 @@
+
+import { ClrTextarea } from './textarea';
+import { ClrTextareaContainer } from './textarea-container';
+import { NgControlService } from '../common/providers/ng-control.service';
+import { Layouts, LayoutService } from '../common/providers/layout.service';
+
+@Component({
+  template: `
+    <clr-textarea-container>
+      <textarea name="test" clrTextarea required [(ngModel)]="model"></textarea>
+      <label>Hello World</label>
+      <clr-control-helper>Helper text</clr-control-helper>
+      <clr-control-error>Must be at least 5 characters</clr-control-error>
+    </clr-textarea-container>
+    `,
+})
+class SimpleTest {
+  model = '';
+}
+
+@Component({
+  template: `
+  <clr-textarea-container>
+    <textarea clrTextarea [(ngModel)]="model"></textarea>
+  </clr-textarea-container>`,
+})
+class NoLabelTest {}
+
+export default function(): void {
+  describe('ClrTextareaContainer', () => {
+    let fixture, textareaContainerDE, textareaContainer, textareaContainerEl, ifErrorService, layoutService;
+
+    describe('no label', () => {
+      beforeEach(() => {
+        TestBed.configureTestingModule({
+          imports: [ClrIconModule, ClrCommonFormsModule, FormsModule],
+          declarations: [ClrTextareaContainer, ClrTextarea, NoLabelTest],
+          providers: [NgControl, NgControlService, IfErrorService, LayoutService],
+        });
+        fixture = TestBed.createComponent(NoLabelTest);
+
+        textareaContainerDE = fixture.debugElement.query(By.directive(ClrTextareaContainer));
+        textareaContainerEl = textareaContainerDE.nativeElement;
+        layoutService = textareaContainerDE.injector.get(LayoutService);
+      });
+
+      it('adds an empty label when instantiated without vertical layout', () => {
+        layoutService.layout = Layouts.HORIZONTAL;
+        fixture.detectChanges();
+        const labels = textareaContainerEl.querySelectorAll('label');
+        expect(Array.prototype.filter.call(labels, label => label.textContent === '').length).toBe(1);
+      });
+    });
+
+    describe('full example', () => {
+      beforeEach(() => {
+        TestBed.configureTestingModule({
+          imports: [ClrIconModule, ClrCommonFormsModule, FormsModule],
+          declarations: [ClrTextareaContainer, ClrTextarea, SimpleTest],
+          providers: [NgControl, NgControlService, IfErrorService, LayoutService],
+        });
+        fixture = TestBed.createComponent(SimpleTest);
+
+        textareaContainerDE = fixture.debugElement.query(By.directive(ClrTextareaContainer));
+        textareaContainer = textareaContainerDE.componentInstance;
+        textareaContainerEl = textareaContainerDE.nativeElement;
+        ifErrorService = textareaContainerDE.injector.get(IfErrorService);
+        layoutService = textareaContainerDE.injector.get(LayoutService);
+        fixture.detectChanges();
+      });
+
+      it('injects the NgControlService and subscribes in AfterContentInit', () => {
```

Comment:
The description is off, it's the `IfErrorService ` there is no `AfterContentInit` involved.

Reply:
Fixed